### DISCUSSION TITLE
Deepscatter / Landscape of Biomedical Research

### DISCUSSION THREAD

**Discussion by sterlinm** [vega/vegafusion#303] [Q&A] [answered]
This [visualization of medical research papers](https://static.nomic.ai/pubmed.html) is pretty cool. Looks like it was created with a project called [Deepscatter](https://github.com/nomic-ai/deepscatter).

I may try to do it myself if I get the time, but it would be interesting to see if it could be recreated with VegaFusion. Could VegaFusion handle 21 million datapoints?

![Screen Shot 2023-04-19 at 10 14 04 AM](https://user-images.githubusercontent.com/1461273/233103118-d2c8a864-1055-4489-b523-4ebcd366c7e7.png)

- The landscape of biomedical research ([paper](https://www.biorxiv.org/content/10.1101/2023.04.10.536208v1), [visualization](https://static.nomic.ai/pubmed.html))
- [Deep Scatterplots for the Web](https://github.com/nomic-ai/deepscatter)

**Comment by jonmmease** [comment, marked as answer]
Hi @sterlinm,

Deepscatter looks like a really neat project. Handling unaggregated scatter plots with millions of points pretty much requires some form of GPU acceleration. Presently, VegaFusion is only involved in the data transformations stage, not the rendering stage. Rendering is still done by the Vega JavaScript library, which doesn't have GPU acceleration yet.  There are a couple of early projects/prototypes to add GPU rendering to Vega ([vega-renderer-webgl](https://github.com/vega/vega-renderer-webgl), [vega-webgpu](https://github.com/vega/vega-webgpu)), but these aren't ready for prime time.

One approach that is sometimes useful is to aggregate x/y data into a heatmap where the bins are approximately the size of one pixel. This is what the VegaFusion [10 Million Taxi Rides](https://vegafusion.io/example_nyc_taxi.html) example does for the middle pickup locations chart. Vega-Lite selections are used to re-aggregate the data on pan and zoom. The downside is that this approach doesn't extend well to include a color encoding, and it doesn't support tooltips.

Working out how to make Deepscatter style charts in Vega is definitely on my wish list, but we're not there yet

**Reply by sterlinm** [reply]
Thanks! To be fair, I have never had an actual need to plot that many points...

I think the sweet spot is somewhere above 5,000 and below 20,000,000 🙂
